### PR TITLE
Cesm component tests4

### PR DESCRIPTION
**Pull request by jedwards4b**
Fix a few more issues found in cesm component tests.  
Test suite:  scripts_regression_tests, several cesm component tests
Test baseline: cesm1_5_alpha06i
Test namelist changes: 
Test status: bit for bit

Fixes:

User interface changes?: no

Code review:

### CODE REVIEW

File: utils/python/CIME/system_test.py [hunk 1]
```diff
@@ -489,24 +496,27 @@ def _xml_phase(self, test):
...
+                            envtest.set_test_parameter(string, opti_tasks)
+                            string = "NTHRDS_" + component_class
+                            envtest.set_test_parameter(string, opti_thrds)
+                            string = "ROOTPE_" + component_class
+                            envtest.set_test_parameter(string, str(0))
+                    elif match3:
+
+                        # handled by create_newcase
+                        continue
+                    if not match3:
```
Comment (jgfouca) [inline utils/python/CIME/system_test.py]: I think you want an "else" here if you want to error-out if nothing matched.

**Comment by jgfouca** [comment]
Minor fix needed before this can be merged.

File: utils/python/CIME/system_test.py (hunk 1 above)
Comment (jedwards4b) [inline utils/python/CIME/system_test.py]: I think that I have it correct - I want to run the expect code in any case except when match3 is true.

Comment (jgfouca) [inline utils/python/CIME/system_test.py]: OK, I see what you're doing now.

**Pull request merged**